protocols/imap/src/main/java/org/apache/james/imap/message/request/GetAnnotationRequest.java

```diff
@@ -58,22 +48,57 @@ public String getMailboxName() {
         return keys;
     }
 
-    public void setKeys(Set<String> keys) {
-        this.keys = ImmutableSet.copyOf(keys);
-    }
-    public long getMaxsizeValue() {
-        return maxsizeValue;
+    public long getMaxsize() {
+        return maxsize;
     }
 
-    public void setMaxsizeValue(long maxsizeValue) {
-        Preconditions.checkArgument(maxsizeValue > 0);
-        this.maxsizeValue = maxsizeValue;
+    public Depth getDepth() {
+        return depth;
     }
 
-    public Depth getDepthValue() {
-        return depthValue;
-    }
+    public static class GetAnnotationRequestBuilder {
+        private final String tag;
+        private final ImapCommand command;
+        private String mailboxName;
+        private Set<String> keys;
+        private long maxsize;
+        private Depth depth;
+
+        public GetAnnotationRequestBuilder(String tag, ImapCommand command) {
+            this.tag = tag;
+            this.command = command;
+            this.depth = Depth.ZERO;
+            keys = ImmutableSet.of();
+        }
+
+        public GetAnnotationRequestBuilder mailboxName(String mailboxName) {
+            Preconditions.checkNotNull(mailboxName);
+            this.mailboxName = mailboxName;
+            return this;
+        }
 
+        public GetAnnotationRequestBuilder keys(Set<String> keys) {
+            this.keys = ImmutableSet.copyOf(keys);
+            return this;
+        }
+
+        public GetAnnotationRequestBuilder maxsize(long maxsize) {
+            Preconditions.checkArgument(maxsize > 0);
+            this.maxsize = maxsize;
+            return this;
+        }
+
+        public GetAnnotationRequestBuilder depth(Depth depth) {
+            this.depth = depth;
+            return this;
+        }
+
+        public GetAnnotationRequest build() {
+            Preconditions.checkArgument(depth == Depth.ZERO || (depth != Depth.ZERO && !keys.isEmpty()));
```

Comment:
name this condition so that my tired, hangovered brain can handle this line :-p

Extract + a if might help